### ISSUE TITLE
gamepad: x, y, a and b buttons does not register within the game

### ISSUE THREAD

**Issue by xyproto** [godotengine/godot#67135]
### Godot version

4.0 beta2

### System information

Arch Linux, GTX 1650, Sway+Wayland, wireless XBox controller

### Issue description

After completing the 2D game tutorial and making adjustments for Godot 4.0, the game works great. The keyboard works and everything is fine.

Adding joypad analog stick and D-pad movement also works great. Perfect. No issues.

Adding keys like space or return to start the game also works great.

However, when adding joypad buttons like `X`, `Y`, `A` and `B`, they do not work within the game, even though they work in the project settings input map window.

### Steps to reproduce

1. Install Godot 4.0 beta 2 on Arch Linux (any Linux distro is probably fine).
3. Have a working project ready, like the 2D tutorial game project.
4. In the "Project Settings" -> "Input Map" window, add the left thumb stick left movement as `move_left`.
5. In the "Project Settings" -> "Input Map" window, add the `A` button on the joypad as `start_game`.
6. Observe that moving the left thumb stick works fine and moves the `Player`, while pressing the `A` button does not start the game, like `space` or `return` does.

### Minimal reproduction project

[tutorial1.zip](https://github.com/godotengine/godot/files/9741765/tutorial1.zip)

**Comment by Calinou** [comment]
> wireless XBox controller

Which exact controller model is this? Are you using it with an USB adapter or Bluetooth? Does this occur in wired mode? Are you using any special driver for wireless (this is often required for good operation on Linux)?

**Comment by xyproto** [comment]
* It's an Xbox One wireless controller (model 1708), like this one:

<img src="https://upload.wikimedia.org/wikipedia/commons/f/ff/Xbox_One_controller_model_1708_%2839160219920%29_%28cropped%29.jpg" width=320>

* I'm using Bluetooth.

* I'll have to test with a wire, but the controller works fine with SDL2 and all sorts of Steam games as it is.

* `cat /dev/input/js0` outputs data as expected when thumbsticks are moved and buttons are pressed.

* I'm not using any special driver for wireless, judging by the `lsmod` output. Also, looking at the `dmesg` output, it appears as a hid-generic input device:

```
13086.368665] hid-generic 0005:045E:02FD.000C: input,hidraw11: BLUETOOTH HID v9.03 Gamepad [Xbox Wireless Controller] on b8:9a:2a:25:2c:ad
[13086.580195] input: Xbox Wireless Controller as /devices/pci0000:00/0000:00:14.0/usb1/1-4/1-4:1.0/bluetooth/hci0/hci0:256/0005:045E:02FD.000C/input/input43
```

In summary:

The joypad works for everything else: Steam, SDL2 and as `/dev/input/js0`. It also works when in the "Project Settings" -> "Input Map" window.

It just doesn't work when running the Godot game.

**Comment by xyproto** [comment]
The project I'm using here (attached further up) was converted from Godot 3.5 to Godot 4.0 beta2, and adjusted to work with Godot 4.0 beta2.

Perhaps this input issue is related to this?

**Comment by xyproto** [comment]
I tested with one of the latest git commits, commit ca25c6e0a3f25948ee4a197f3442c66f019e7424.

It works there.

**Comment by xyproto** [comment]
If it works in one of the latest commits, but not in `4.0 beta2`, should I leave this issue open or close it?

**Comment by Calinou** [comment]
> If it works in one of the latest commits, but not in `4.0 beta2`, should I leave this issue open or close it?

This may be related to udev being compiled in official builds, but not in self-compiled builds if you don't have it installed. (udev is used for controller hotplugging support.)

**Comment by xyproto** [comment]
I'm maintaining the `godot` package for Arch Linux. The dependencies when I tested `beta 2` are the same as for the official package, and the packages are built in a chroot-ed environment. I might be wrong, but I don't think the issue is udev related.

**Comment by xyproto** [comment]
This now works perfectly in 276ab5be07752582b4341c33fb8f53a7887af87a (probably included in `4.0 beta 3`).
